p2p/src/error.rs

```diff
@@ -14,124 +14,133 @@
 // limitations under the License.
 //
 // Author(s): A. Altonen
+use libp2p::{
+    gossipsub::error::PublishError as GossipsubPublishError,
+    swarm::{handler::ConnectionHandlerUpgrErr, DialError::*},
+};
 use thiserror::Error;
 
-// TODO: think about which errors should be returned and when
-// TODO: store peerid where appropriate!
 #[derive(Error, Debug, PartialEq, Eq)]
 pub enum ProtocolError {
+    #[error("Peer is in different network")]
     DifferentNetwork,
+    #[error("Peer has an unsupported version")]
     InvalidVersion,
+    #[error("Peer sent an invalid message")]
     InvalidMessage,
+    #[error("Peer is incompatible")] // TODO: remove?
     Incompatible,
+    #[error("Peer is unresponsive")]
```

Comment:
Perhaps add the definition of Unresponsive in the error text

Reply:
I removed the error, it's not used anywhere. I'll add it back if it's needed